dandischema/metadata.py

```diff
@@ -56,29 +56,34 @@ def generate_context() -> dict:
     fields: Dict[str, Any] = {}
     for val in dir(models):
         klass = getattr(models, val)
-        if not isinstance(klass, pydantic.main.ModelMetaclass):
+        if not isinstance(klass, pydantic._internal._model_construction.ModelMetaclass):
             continue
         if hasattr(klass, "_ldmeta"):
-            if "nskey" in klass._ldmeta:
+            if "nskey" in klass._ldmeta.default:
                 name = klass.__name__
-                fields[name] = f'{klass._ldmeta["nskey"]}:{name}'
-        for name, field in klass.__fields__.items():
+                fields[name] = f'{klass._ldmeta.default["nskey"]}:{name}'
+        for name, field in klass.model_fields.items():
             if name == "id":
                 fields[name] = "@id"
             elif name == "schemaKey":
                 fields[name] = "@type"
             elif name == "digest":
                 fields[name] = "@nest"
             elif name not in fields:
-                if "nskey" in field.field_info.extra:
-                    fields[name] = {"@id": field.field_info.extra["nskey"] + ":" + name}
+                if (
+                    field.json_schema_extra is not None
+                    and "nskey" in field.json_schema_extra
+                ):
+                    fields[name] = {
+                        "@id": field.json_schema_extra["nskey"] + ":" + name
+                    }
                 else:
                     fields[name] = {"@id": "dandi:" + name}
-                if "List" in str(field.outer_type_):
+                if "List" in str(field.annotation):
                     fields[name]["@container"] = "@set"
                 if name == "contributor":
                     fields[name]["@container"] = "@list"
-                if "enum" in str(field.type_) or name in ["url", "hasMember"]:
+                if "enum" in str(field.annotation) or name in ["url", "hasMember"]:
```

Comment:
I suspect the first check can be implemented better by inspecting the actual annotation instead of stringifying it, but I'm not clear on what exactly it's testing for.  Quickly looking over `models.py`, I don't see anything with "enum" in an annotation.  CC @satra

Reply:
i would check pydantic v1 generate context and break there to see what field it was picking up. if it's not relevant anymore, then fine to replace or remove.

Reply:
It picks a few fields. Among them are `AcccessRequirements.status` and `Asset.license`.

I have also updated the code to not use the method of stringifying.